python/ct/client/db/cert_desc_test.py

```diff
@@ -13,81 +14,86 @@
-            self.assertItemsEqual(cert_subject, subject)
-
-            issuer = [(att.type, att.value) for att in proto.issuer]
-            cert_issuer = [(type_.short_name,
-                         cert_desc.to_unicode('.'.join(
-                                 cert_desc.process_name(value.human_readable()))))
-                        for type_, value in source.issuer()]
-            self.assertItemsEqual(cert_issuer, issuer)
-
-            subject_alternative_names = [(att.type, att.value)
-                                         for att in proto.subject_alternative_names]
-            cert_subject_alternative_names = [(san.component_key(),
-                                               cert_desc.to_unicode('.'.join(
-                                                cert_desc.process_name(
-                                         san.component_value().human_readable()))))
-                        for san in source.subject_alternative_names()]
-            self.assertItemsEqual(cert_subject_alternative_names,
-                                  subject_alternative_names)
-
-            self.assertEqual(proto.version, str(source.version().value))
-            self.assertEqual(proto.serial_number,
-                             str(source.serial_number().human_readable()
-                                 .upper().replace(':', '')))
-            self.assertEqual(time.gmtime(proto.validity.not_before / 1000),
-                             source.not_before())
-            self.assertEqual(time.gmtime(proto.validity.not_after / 1000),
-                             source.not_after())
-
-            observations_tuples = [(unicode(obs.description),
-                                    unicode(obs.reason) if obs.reason else u'',
-                                    obs.details_to_proto())
-                                   for obs in observations]
-            proto_obs = [(obs.description, obs.reason, obs.details)
-                         for obs in proto.observations]
-            self.assertItemsEqual(proto_obs, observations_tuples)
-
-            self.assertEqual(proto.tbs_signature.algorithm_id,
-                             source.signature()["algorithm"].long_name)
-            self.assertEqual(proto.cert_signature.algorithm_id,
-                             source.signature_algorithm()["algorithm"].long_name)
-            self.assertEqual(proto.tbs_signature.algorithm_id,
-                             proto.cert_signature.algorithm_id)
-
-            if source.signature()["parameters"]:
-                self.assertEqual(proto.tbs_signature.parameters,
-                                 source.signature()["parameters"])
-            else:
-                self.assertFalse(proto.tbs_signature.HasField('parameters'))
+        # Test in a non-UTC timezone, to detect timezone issues
+        with time_utils.timezone("America/Los_Angeles"):
+            for test_case in [(CERT, False), (DSA_SHA256_CERT, False), (CA_CERT, True)]:
```

Comment:
This test method is getting rather substantial. It has a 'with' clause, a for clause (for 3 different cases), a couple of if/else statements and quite a bit of code in between. How about breaking it up?